File: mobatech-flutter/lib/features/home/data/models/branch.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'branch.freezed.dart';
part 'branch.g.dart';

@freezed
abstract class Branch with _$Branch {
  factory Branch({
    @JsonKey(name: 'ID', defaultValue: 0) required int id,
    @JsonKey(defaultValue: '') required String name,
    @JsonKey(defaultValue: '') required String address,
    @JsonKey(defaultValue: 0.0) required double latitude,
    @JsonKey(defaultValue: 0.0) required double longitude,
    @JsonKey(name: 'image_url', defaultValue: '') required String imageUrl,
    @JsonKey(name: 'gmaps_link', defaultValue: '') required String gmapsLink,
  }) = _Branch;

  factory Branch.fromJson(Map<String, dynamic> json) => _$BranchFromJson(json);
}

File: mobatech-backend/models/branch.go
package models

import "gorm.io/gorm"

type Branch struct {
	gorm.Model
	Name      string  `json:"name"`
	Address   string  `json:"address"`
	Latitude  float64 `json:"latitude"`
	Longitude float64 `json:"longitude"`
	ImageURL  string  `json:"image_url"`
	GmapsLink string  `json:"gmaps_link"`
}
